client/my-sites/google-my-business/connect-button/index.js

```diff
@@ -0,0 +1,135 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { last, isEqual } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import { deleteStoredKeyringConnection } from 'state/sharing/keyring/actions';
+import { SharingService, connectFor } from 'my-sites/sharing/connections/service';
+import { getKeyringServiceByName } from 'state/sharing/services/selectors';
+import QueryKeyringServices from 'components/data/query-keyring-services';
+
+class GoogleMyBusinessConnectButton extends SharingService {
+	static propTypes = {
+		...SharingService.propTypes,
+		deleteStoredKeyringConnection: PropTypes.func,
+		onClick: PropTypes.func,
+		onConnect: PropTypes.func,
+	};
+
+	static defaultProps = {
+		...SharingService.defaultProps,
+		deleteStoredKeyringConnection: () => {},
+		onClick: () => {},
+		onConnect: () => {},
+	};
+
+	// override `createOrUpdateConnection` to ignore connection update, this is only useful for publicize services
+	createOrUpdateConnection = () => {};
+
+	// override `removeConnection` to remove the keyring connection instead of the publicize one
+	removeConnection = () => {
+		this.setState( { isDisconnecting: true } );
+		this.props.deleteStoredKeyringConnection( last( this.props.keyringConnections ) );
+	};
+
+	onClick = () => {
+		this.performAction();
+		this.props.onClick();
```

Comment:
Shouldn't we wait for `performAction` to complete connection before calling `onClick`?

Reply:
I see, there's `onConnect` that will be called when the connection established. sorry.